src/components/input-field/input-field.tsx

```diff
@@ -292,32 +295,35 @@ export class InputField {
         properties.readonly = this.readonly;
         properties.disabled = this.disabled || this.readonly;
 
+        if (this.hasHelperText()) {
+            properties['aria-controls'] = helperTextId;
+            properties['aria-describedby'] = helperTextId;
+        }
+
         const labelClassList = {
             'mdc-floating-label': true,
             'mdc-floating-label--float-above':
                 !!this.value || this.isFocused || this.readonly,
         };
 
         return [
-            <div class={this.getContainerClassList()}>
+            <label class={this.getContainerClassList()}>
+                <span class="mdc-notched-outline">
+                    <span class="mdc-notched-outline__leading"></span>
+                    <span class="mdc-notched-outline__notch">
+                        <span class={labelClassList} id={labelId}>
+                            {this.label}
+                        </span>
+                    </span>
+                    <span class="mdc-notched-outline__trailing"></span>
+                </span>
+                {this.renderLeadingIcon()}
                 {this.renderFormattedNumber()}
                 {this.renderEmptyValueForReadonly()}
                 {this.renderInput(properties)}
                 {this.renderTextarea(properties)}
-                <div class="mdc-notched-outline">
-                    <div class="mdc-notched-outline__leading"></div>
-                    <div class="mdc-notched-outline__notch">
-                        <label
-                            class={labelClassList}
-                            htmlFor="tf-input-element"
-                        >
-                            {this.label}
-                        </label>
-                    </div>
-                    <div class="mdc-notched-outline__trailing"></div>
-                </div>
-                {this.renderIcons()}
-            </div>,
+                {this.renderTrailingIconNew()}
```

Comment:
leading and trailing icons should be placed and rendered before `mdc-notched-outline`, otherwise they create styling issues.

![image](https://user-images.githubusercontent.com/35954987/125780045-6b4be24a-94de-4fad-80da-c975ee95448c.png)

Reply:
I'm not entirely convinced that the code structure used on material.io should take precedence over the code structure used in the docs for the MDC version we are using. But if something isn't working as it should, and you can make it work by moving stuff around, feel free to do so 😄

Reply:
>  those trailing links (phone, email, website) cannot be clicked, since they will be covered by other stuff.
Me neither. But this is one of the problems with it. You can test it in `readonly` mode.

Reply:
Well, let's go with the code structure of MDC. In this case, I can say that since those trailing and leading icons are missing styles (which gives them `position: absolute`), it doesn't matter if they are first in the DOM or not. They should be visually rendered on a layer below the value, and also below the `formatted-input` span. Otherwise, our styles in `readonly` for `showLink=true` example won't work correctly. I push a fixup now. Can you please see how we can get those missing styles that [I explained here](https://github.com/Lundalogik/lime-elements/pull/1297#discussion_r670399736)? I can't figure out how.

But I should say it's impossible to track down visual issue, without a rendered example. They link to material.io from MDC docs on github :/ to showcase the components.

Reply:
> They should be visually rendered on a layer below the value, and also below the `formatted-input` span.

I'm pretty sure they changed that. They're now using flex to handle the placing of the icons and the input element, which is why the dom structure has changed. We can override any styling we want in the readonly styles, but I don't love the idea of overriding the basic styles to allow for a completely different dom structure, as I think that would likely be a constant source of future suffering 😄

Can't we update our readonly styles to fit the new structure instead?

Reply:
> I don't love the idea of overriding the basic styles to allow for a completely different dom structure

I hate this idea 😂

cd> Can't we update our readonly styles to fit the new structure instead?

We certainly can. I was not aware that material.io is different than MDC. Well, if we load all their styles, then we should just modify our own modifications. 


> They're now using flex to handle the placing of the icons and the input element, 

Hm.... OK. Let me check.

Reply:
Can't we view some vanilla MDCs somewhere?

Reply:
I can probably set something just using what they describe in the docs… I'll have a look at that tomorrow, if you remind me 😄

Reply:
It'd be awesome. Just to be able to compare how their vanilla components work, without any of our customizations. It's easier to track missing styles, and more.

Reply:
Yeah, I'll do that once all components are "not-broken". We shouldn't be spending time on polishing before then anyway. See it as a carrot, the less time you spend on polishing now, the faster we can get everything to "not-broken", and then I'll look at getting you that vanilla version to use while polishing. Ok? 😊